gensim/summarization/graph.py

```diff
@@ -303,31 +289,27 @@ def add_edge(self, edge, wt=1, label='', attrs=None):
             Given edge.
         wt : float, optional
             Weight of new edge.
-        label : str, optional
-            Edge label.
-        attrs : list, optional
-            Node attributes specified as (attribute, value).
 
         Raises
         ------
         ValueError
             If `edge` already exists in graph.
 
         """
-        if attrs is None:
-            attrs = []
+        if wt == 0.0:
+            # empty edge is similar to no edge at all or removing it
+            if self.has_edge(edge):
+                self.del_edge(edge)
+            return
         u, v = edge
         if v not in self.node_neighbors[u] and u not in self.node_neighbors[v]:
-            self.node_neighbors[u].append(v)
+            self.node_neighbors[u][v] = wt
             if u != v:
-                self.node_neighbors[v].append(u)
-
-            self.add_edge_attributes((u, v), attrs)
-            self.set_edge_properties((u, v), label=label, weight=wt)
+                self.node_neighbors[v][u] = wt
         else:
             raise ValueError("Edge (%s, %s) already in graph" % (u, v))
 
-    def add_node(self, node, attrs=None):
+    def add_node(self, node):
```

Comment:
we need only weight always and for this reason, you get rid attrs, right?

Reply:
Yes, even in original commit of @fbarrios this attrs was not used anywhere else besides Graph itself. Labels in original commits was used in one script that already deleted anyway. 
I don't like that I broke backward compatibility but on the other side gensim is not a library for the graphs and it's an inner class. I may like a compromise between compatibility and efficiency create something like a SimpleGraph and use it and deprecate ordinary Graph.

Reply:
That's fine I guess, this is some stuff for internal usage, in this case, we shouldn't get any problems with it